### PR TITLE
statistics should exclude teachers and tutors

### LINKED ISSUE #600: Statistics should exclude teachers and tutors.
... as it distorts the information.

### PR DESCRIPTION
**Pull request by Jonas960**
resolves #600

### CODE REVIEW

**Review by Zitrone44 — CHANGES_REQUESTED** [review]
Works good :+1:

But i would recommend reducing repetition by dynamically building the query instead of completely copying it for `getStudResult`.

**Comment by Jonas960** [comment]
@Zitrone44 The query is now dynamically generated. The query must now be called with the cid and the role numbers (from, to).

File: ws/src/main/scala/de/thm/ii/fbs/services/persistence/CourseResultService.scala [hunk 1]
```diff
@@ -28,7 +28,7 @@ class CourseResultService {
     * @param cid Course id
     * @return all Course Results
     */
-  def getAll(cid: Int): List[CourseResult] = DB.query("""
+  def getAll(cid: Int, from: Int, to: Int): List[CourseResult] = DB.query("""
```
Comment (Zitrone44) [inline ws/src/main/scala/de/thm/ii/fbs/services/persistence/CourseResultService.scala:31]:
```suggestion
  def getAll(cid: Int, from: Int = 0, to: Int = 2): List[CourseResult] = DB.query("""
```

File: ws/src/main/scala/de/thm/ii/fbs/controller/CourseResultController.scala [hunk 2]
```diff
@@ -66,9 +66,31 @@ class CourseResultController {
       .exists(p => p.role == CourseRole.DOCENT || p.role == CourseRole.TUTOR)
 
     if (privilegedByCourse || user.globalRole == GlobalRole.ADMIN || user.globalRole == GlobalRole.MODERATOR) {
-      evaluationResultService.evaluate(evaluationContainerService.getAll(cid), courseResultService.getAll(cid))
+      evaluationResultService.evaluate(evaluationContainerService.getAll(cid), courseResultService.getAll(cid, 0, 2))
```
Comment (Zitrone44) [inline ws/src/main/scala/de/thm/ii/fbs/controller/CourseResultController.scala:69]:
```suggestion
      evaluationResultService.evaluate(evaluationContainerService.getAll(cid), courseResultService.getAll(cid))
```

File: ws/src/main/scala/de/thm/ii/fbs/controller/CourseResultController.scala [hunk 3]
```diff
@@ -44,7 +44,7 @@ class CourseResultController {
       .exists(p => p.role == CourseRole.DOCENT || p.role == CourseRole.TUTOR)
 
     if (privilegedByCourse || user.globalRole == GlobalRole.ADMIN || user.globalRole == GlobalRole.MODERATOR) {
-      courseResultService.getAll(cid)
+      courseResultService.getAll(cid, 0, 2)
```
Comment (Zitrone44) [inline ws/src/main/scala/de/thm/ii/fbs/controller/CourseResultController.scala:47]:
```suggestion
      courseResultService.getAll(cid)
```

**Review by Zitrone44 — CHANGES_REQUESTED** [review]
Like @mxsph suggested at the meeting default values should be added to `getAll`.

**Review by Zitrone44 — APPROVED** [review]

**Pull request merged**